homeassistant/components/onvif/camera.py

```diff
@@ -201,29 +158,39 @@ async def async_initialize(self):
         the camera. Also retrieves the ONVIF profiles.
         """
         try:
-            _LOGGER.debug("Updating service addresses")
+            LOGGER.debug("Updating service addresses")
             await self._camera.update_xaddrs()
 
+            await self.async_obtain_device_info()
             await self.async_obtain_mac_address()
             await self.async_check_date_and_time()
+            await self.async_obtain_profile_token()
             await self.async_obtain_input_uri()
             await self.async_obtain_snapshot_uri()
```

Comment:
How much of this data is profile specific and how much data can be shared between all entities of 1 camera?

Reply:
The very next PR is going to refactor almost all of this logic out of the Camera entity and into a Device class, which will make it easier to implement events/sensors as well. I wanted to limit the scope of this PR though.